index.js

```diff
@@ -597,11 +597,13 @@ async function apostrophe(options, telemetry, rootSpan) {
 
     // The base module sort metadata
     for (const name of moduleNames) {
-      if (definitions[name].before) {
+      const metadata = await self.synth.getMetadata(name);
+      const before = Object.values(metadata.before).reverse().find(name => typeof name === 'string');
```

Comment:
Good call on doing `reverse` here so a subclass can have its own `before` if it does not make sense to inherit `before` in a particular use case.